app/main/forms/suppliers.py

```diff
@@ -139,3 +140,23 @@ class CompanyOrganisationSizeForm(Form):
     organisation_size = RadioField('Organisation size',
                                    validators=[InputRequired(message="You must choose an organisation size.")],
                                    choices=[(o['value'], o['label']) for o in OPTIONS])
+
+
+class VatNumberForm(Form):
+    def stop_validation_if_not_registered(form, field):
+        # If a user is not registered for VAT we don't care about validating something they have entered in the
+        # VAT number input field. If a value for `vat_registered` is not sent in the POST request, WTForms sets the
+        # value to `None`, hence its inclusion here
+        if form.vat_registered.data in ('No', 'None'):
+            raise StopValidation()
+
+    vat_registered = RadioField("VAT registered",
+                                validators=[InputRequired(message="You need to answer this question.")],
+                                choices=[('Yes', 'Yes'), ('No', 'No')])
+    vat_number = StripWhitespaceStringField('VAT number', default="", validators=[
+        stop_validation_if_not_registered,
+        InputRequired(message="You must provide a VAT number."),
+        Regexp(r'^([GB])*((\d{9})|(\d{12})|((GD|HA)\d{3}))$',
+               flags=IGNORECASE,
```

Comment:
should we make all numbers either uppercase or lowercase before we save them into the database?

Reply:
Good call. I've updated to uppercase everything.